src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -225,4 +226,16 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig, protected va
       case None => params
     }
   }
-}
+
+  private[service] def addClusterDefaultLabels(cluster: Cluster): Cluster = {
+    // create a LabelMap of default labels
+    val defaultLabels = DefaultLabels(cluster.clusterName, cluster.googleProject, cluster.googleBucket, cluster.googleServiceAccount, cluster.jupyterExtensionUri)
+      .toJson.asJsObject.fields.mapValues(labelValue => labelValue.convertTo[String])
+    // combine default and given labels
+    val allLabels = cluster.labels ++ defaultLabels
+    // check the labels do not contain forbidden keys
+    if (allLabels.contains(includeDeletedKey))
+      throw IllegalLabelKeyException(includeDeletedKey)
```

Comment:
As it is now, this will be thrown after the dataproc cluster is already created in Google - this check should happen before that

Reply:
Right, will fix